### PR TITLE
Fix issue 636, pass request object to authenticate function.

### PR DESCRIPTION
**Pull request by W0126**
Pass request object to authenticate function,  request may used in authenticate function.

### CODE REVIEW

File: oauth2_provider/oauth2_validators.py [hunk 1]
```diff
@@ -572,7 +572,7 @@ def validate_user(self, username, password, client, request, *args, **kwargs):
         """
         Check username and password correspond to a valid and active User
         """
-        u = authenticate(username=username, password=password)
+        u = authenticate(request, username=username, password=password)
         if u is not None and u.is_active:
```
Comment (cleder) [inline oauth2_provider/oauth2_validators.py]: `is_active` is already checked in django (>=1.11 I think) do we need to check this here again?

Comment (W0126) [reply oauth2_provider/oauth2_validators.py]: This fix make sure request as a parameter to authenticate function, so authentication module can get request information.

**Comment by adamchainz** [comment]
This would be useful for me. I just discovered django-axes, *requires* `request` to be passed, and throws an exception if it's not: https://django-axes.readthedocs.io/en/latest/3_usage.html#authenticating-users

What's the blocker on this getting merged? I'm happy to take over.

**Comment by W0126** [comment]
@adamchainz  we may need at least 1 reviewers with write access to approve this.

**Comment by adamchainz** [comment]
@cleder ? Also I volunteer myself to be given write access if that helps :)

File: oauth2_provider/oauth2_validators.py (hunk 1 above)
Comment (IvanAnishchuk) [reply oauth2_provider/oauth2_validators.py]: @cleder I agree regarding `is_active`, in fact Django also skips the check if user model doesn't implement `is_active`. Let's remove this part.

Comment (IvanAnishchuk) [reply oauth2_provider/oauth2_validators.py]: It could be a separate PR though. Approving as the fix is correct.

**Comment by adamchainz** [comment]
Please revert this, it's passing a non django request to `authenticate()` !

**Comment by Nocturn78** [comment]
> Please revert this, it's passing a non django request to `authenticate()` !

Exactly

**Comment by n2ygk** [comment]
I will try to look at this over the weekend. Unfortunately the project
settings at the time allowed a merge without review although I’m not sure
another reviewer would have noticed as the tests passed. A PR with a
failing test that shows the error and fixes it by reverting would be
appreciated.

Updated contributing guidelines hopefully reinforce why good test coverage
is needed:
https://django-oauth-toolkit.readthedocs.io/en/latest/contributing.html

**Comment by indigodomo** [comment]
When is this going to get updated in pypi? The broken version is in there now. I really don't want to have to deploy with my own copy...

**Comment by n2ygk** [comment]
You can always use a git reference in your requirements.txt. For example:
```
git+https://github.com/jazzband/django-oauth-toolkit.git@9bd4c1bfcf26eecf8fde793baa48c386885c19be
```

**Comment by adamchainz** [comment]
Or install from a commit has on github: https://adamj.eu/tech/2019/03/11/pip-install-from-a-git-repository/

**Pull request merged**